### ISSUE TITLE
Rumble Support

### ISSUE THREAD

**Issue by Noah-Steinberg** [paroj/xpad#67]
Is there any rumble support for any controllers using xpad?
I am currently working on my own driver that supports rumble for the microsoft xbox one controllers, and if it is wanted, I could look at doing it in xpad as well. If I had the time, I could look into other controllers as well.

**Comment by cgutman** [comment]
Yes, rumble is supported for all gamepads that xpad supports: Xbox, Xbox 360 (Wired and Wireless), and Xbox One (Wired).

See: https://github.com/paroj/xpad/blob/master/xpad.c#L1013

**Comment by Noah-Steinberg** [comment]
Ah, I see, I had not looked closely enough at xpad. Thanks!

**Comment by Noah-Steinberg** [comment]
Is there any wish for more rumble support than a single packet type with strong/weak?

**Comment by cgutman** [comment]
You mean support rumble on the triggers? or something else?

Supporting trigger rumble is trivial from an xpad perspective, but I couldn't find a sensible way to map to the force-feedback API in the kernel.

**Comment by Noah-Steinberg** [comment]
I am currently looking into any way to do it. The only issue is having to construct the packet byte by byte. So far I have support for the different kinds of rumble that the kernel specifies, but in very primitive and mostly hard coded ways. Also, Im getting all my information from the Xbox One Controller Protocall
